gson/src/main/java/com/google/gson/internal/bind/TypeAdapters.java

```diff
@@ -788,7 +794,11 @@ public EnumTypeAdapter(Class<T> classOfT) {
           nameToConstant.put(name, constant);
           constantToName.put(constant, name);
         }
-      } catch (NoSuchFieldException e) {
+      } catch (IllegalAccessException e) {
+        throw new AssertionError(e);
+      } catch (NullPointerException e) {
+        throw new AssertionError(e);
+      } catch (ExceptionInInitializerError e) {
```

Comment:
Why catch this?

Reply:
For https://docs.oracle.com/javase/7/docs/api/java/lang/reflect/Field.html#get(java.lang.Object)

Reply:
Yes, but only because the method can throw the error does not mean that it should be caught here. I assume if that error actually occurred it could also happen in the non-Proguard specific code and there it is not caught either. So in my opinion it should not be caught here.

(Though note that I am not a maintainer of this project)

Reply:
Fixed at eba1415f1c20a225e528aac83913c3a09962c282